dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -338,91 +348,95 @@ public Group createWorkflowGroup(Context context, Collection collection, int ste
         // Check authorisation - Must be an Admin to create Workflow Group
         AuthorizeUtil.authorizeManageWorkflowsGroup(context, collection);
 
-        if (getWorkflowGroup(collection, step) == null) {
+        if (getWorkflowGroup(context, collection, step) == null) {
             //turn off authorization so that Collection Admins can create Collection Workflow Groups
             context.turnOffAuthorisationSystem();
             Group g = groupService.create(context);
-            context.restoreAuthSystemState();
-
             groupService.setName(g,
                                  "COLLECTION_" + collection.getID() + "_WORKFLOW_STEP_" + step);
             groupService.update(context, g);
+            context.restoreAuthSystemState();
             setWorkflowGroup(context, collection, step, g);
-
         }
 
-        return getWorkflowGroup(collection, step);
+        return getWorkflowGroup(context, collection, step);
     }
 
     @Override
     public void setWorkflowGroup(Context context, Collection collection, int step, Group group)
         throws SQLException, AuthorizeException {
-        // we need to store the old group to be able to revoke permissions if granted before
-        Group oldGroup = null;
-        int action;
+        Workflow workflow = null;
+        try {
+            workflow = workflowFactory.getWorkflow(collection);
+        } catch (IOException | WorkflowConfigurationException e) {
+            log.error(LogManager.getHeader(context, "setWorkflowGroup",
+                    "collection_id=" + collection.getID() + " " + e.getMessage()), e);
+        }
+        if (!StringUtils.equals("default", workflow.getID())) {
+            throw new IllegalArgumentException(
+                    "setWorkflowGroup can be used only on collection with the default basic dspace workflow. "
+                    + "Instead, the collection: "
+                            + collection.getID() + " has the workflow: " + workflow.getID());
+        }
+        String roleId;
 
         switch (step) {
             case 1:
-                oldGroup = collection.getWorkflowStep1();
-                action = Constants.WORKFLOW_STEP_1;
-                collection.setWorkflowStep1(group);
+                roleId = "reviewer";
```

Comment:
All these `roleId` options look like they should possibly be constants as well.  I'm unclear why they are hardcoded (both in this method and in `getWorkflowGroup()`

Reply:
I will move these string to constant. They are hardcoded because the method on the collectionService refers to the "legacy workflow". I have wrapped these call to the equivalent "configuration" to simplify porting of other features that rely on the legacy workflow

Reply:
done